metrics/e2e_test.go

```diff
@@ -0,0 +1,629 @@
+/*
+Copyright 2020 The Knative Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package metrics
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"net"
+	"net/http"
+	"os"
+	"sort"
+	"strings"
+	"sync"
+	"testing"
+	"time"
+
+	sd "contrib.go.opencensus.io/exporter/stackdriver"
+	ocmetrics "github.com/census-instrumentation/opencensus-proto/gen-go/agent/metrics/v1"
+	ocresource "github.com/census-instrumentation/opencensus-proto/gen-go/resource/v1"
+	"go.opencensus.io/resource"
+	"go.opencensus.io/stats"
+	"go.opencensus.io/stats/view"
+	"go.opencensus.io/tag"
+	"k8s.io/apimachinery/pkg/util/wait"
+
+	emptypb "github.com/golang/protobuf/ptypes/empty"
+	"github.com/google/go-cmp/cmp"
+	"google.golang.org/api/option"
+	metricpb "google.golang.org/genproto/googleapis/api/metric"
+	stackdriverpb "google.golang.org/genproto/googleapis/monitoring/v3"
+	"google.golang.org/grpc"
+	proto "google.golang.org/protobuf/proto"
+
+	logtesting "knative.dev/pkg/logging/testing"
+	"knative.dev/pkg/metrics/metricskey"
+	"knative.dev/pkg/metrics/metricstest"
+)
+
+var (
+	NamespaceTagKey = tag.MustNewKey(metricskey.LabelNamespaceName)
+	ServiceTagKey   = tag.MustNewKey(metricskey.LabelServiceName)
+	ConfigTagKey    = tag.MustNewKey(metricskey.LabelConfigurationName)
+	RevisionTagKey  = tag.MustNewKey(metricskey.LabelRevisionName)
+)
+
+type metricExtract struct {
+	Name   string
+	Labels map[string]string
+	Value  int64
+}
+
+func (m metricExtract) Key() string {
+	return fmt.Sprintf("%s<%s>", m.Name, resource.EncodeLabels(m.Labels))
+}
+
+func (m metricExtract) String() string {
+	return fmt.Sprintf("%s:%d", m.Key(), m.Value)
+}
+
+func initSdFake(sdFake *stackDriverFake) error {
```

Comment:
I guess?
```suggestion
func initSDFake(sdFake *stackDriverFake) error {
```

Reply:
Done. I have no idea, the product name is "Stackdriver", so I just spelled it out the 3 places it is used.